### DISCUSSION TITLE
Auto Discovery für das RM-HA-AddOn

### DISCUSSION THREAD

**Discussion by Baxxy13** [SukramJ/aiohomematic#257] [Q&A]
Alle meine externen RM-Testsysteme werden per Auto Discovery gefunden und lassen sich problemlos einbinden.

Das RM-HA-AddOn was auf einem meiner beiden Test-HA-VM's läuft aber nicht.
Da ich recht unwissend in Sachen Containerisierung sowie "Ports nach außen leiten" bin, dazu ein paar Fragen.

Ist es technisch möglich das Auto Discovery auch das RM-HA-AddOn erkennt?
Welchen Port müsste man "nach außen leiten"?

**Comment by SukramJ** [comment]
Also die Verwendung der Discoverfunktion in HA ist so einfach, das ich mir dazu keine Gedanken machen musste.
Laienhafte Darstellung:
Vom Grundsatz ist es so, das HA einen UPNP-Broadcast empfängt, und in der Integration nach einer Filterung auf den korrekten Hersteller die empfangenen Daten auswertet. 
Ansonsten kann man die Statusseite über http://x.x.x.x/upnp/basic_dev.cgi aufrufen. Siehe issue #224.

Das RM-HA-AddOn läuft in einem Docker-Container in einem Docker-Netz (kein Host-Netzwerk) so der Broadcast erst mal nicht nach draussen gelangt. Was ich noch nicht geprüft habe ist, ob RM und HA im gleichen Docker-Netzwerk laufen, dann sollte in dem Fall das Auto-Discovery funktionieren.

Mehr Details bei [Wikipedia](https://de.wikipedia.org/wiki/Universal_Plug_and_Play#Lokalisierung_(Discovery)).

**Comment by SukramJ** [comment]
HA und RM laufen beide im `hassio` Docker Netzwerk.
Im Add-On kann man den Port 1900/UDP (ssdpd auto-discovery (Extern)) freischalten.
Aber beides ist ist nicht erfolgreich.

Man müsste jetzt mit einem UPNP-Tools (z. Bsp. iPhone Net Analyzer) schauen ob UPNP erkannt wird, aber da ich gerade nicht zuhause bin kann ich das heute jetzt nicht testen.

**Comment by noxhirsch** [comment]
Möglicherweise hängt das mit diesem Problem zusammen: https://github.com/jens-maus/RaspberryMatic/issues/1373#issuecomment-1021092709
Hier geht es allerdings um das Finden der CCU von außerhalb des Docker Netzwerkes - also kann gut sein, dass das nicht damit zusammenhängt.
Jens Maus wollte das Problem demnächst angehen, wenn die nächste HAOS-Version released wurde, die ein vorheriges Problem löst (dass für das Addon bisher der Protection Mode deaktiviert werden musste)

**Reply by SukramJ** [reply]
Naja es geht darum wie Multicast Traffic in den Container kommt, bzw. aus den Container heraus gelangt.
Wenn das gelöst ist sollte eigentlich auch das Discovery funktionieren.

**Comment by Baxxy13** [comment]
Ah ja, danke erstmal für die Informationen.
Dann warten wir mal ab und schauen ob sich da bei Gelegenheit was tut.
